commit 4302e5d53b9166d45317e3ddf0a7a9dab3efd43b
Author: Ralf Baechle <ralf@linux-mips.org>
Date:   Thu Mar 5 11:45:48 2009 +0100

    MIPS: compat: Implement is_compat_task.
    
    This is a build fix required after "x86-64: seccomp: fix 32/64 syscall
    hole" (commit 5b1017404aea6d2e552e991b3fd814d839e9cd67).  MIPS doesn't
    have the issue that was fixed for x86-64 by that patch.
    
    This also doesn't solve the N32 issue which is that N32 seccomp processes
    will be treated as non-compat processes thus only have access to N64
    syscalls.
    
    Signed-off-by: Ralf Baechle <ralf@linux-mips.org>
    Signed-off-by: Linus Torvalds <torvalds@linux-foundation.org>

Backported to Debian's 2.6.26 by dann frazier <dannf@debian.org>

diff -urpN linux-source-2.6.26.orig/include/asm-mips/compat.h linux-source-2.6.26/include/asm-mips/compat.h
--- linux-source-2.6.26.orig/include/asm-mips/compat.h	2008-07-13 15:51:29.000000000 -0600
+++ linux-source-2.6.26/include/asm-mips/compat.h	2009-05-06 08:16:58.000000000 -0600
@@ -3,6 +3,8 @@
 /*
  * Architecture specific compatibility types
  */
+#include <linux/seccomp.h>
+#include <linux/thread_info.h>
 #include <linux/types.h>
 #include <asm/page.h>
 #include <asm/ptrace.h>
@@ -218,4 +220,9 @@ struct compat_shmid64_ds {
 	compat_ulong_t	__unused2;
 };
 
+static inline int is_compat_task(void)
+{
+	return test_thread_flag(TIF_32BIT);
+}
+
 #endif /* _ASM_COMPAT_H */
